homeassistant/components/matter/cover.py

```diff
@@ -0,0 +1,157 @@
+        """Return if the cover is closing or not."""
+        operational_status = self.get_matter_attribute_value(
+            clusters.WindowCovering.Attributes.OperationalStatus
+        )
+
+        assert operational_status is not None
+
+        LOGGER.debug(
+            "GOT OPERATIONAL STATUS %s for %s", operational_status, self.entity_id
+        )
+        state = operational_status & 0b11
+        return state == 0b10
+
+    @property
+    def is_opening(self) -> bool:
+        """Return if the cover is opening or not."""
+        operational_status = self.get_matter_attribute_value(
+            clusters.WindowCovering.Attributes.OperationalStatus
+        )
+
+        assert operational_status is not None
+
+        LOGGER.debug(
+            "GOT OPERATIONAL STATUS %s for %s", operational_status, self.entity_id
+        )
+        state = operational_status & 0b11
+        return state == 0b01
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover movement."""
+        await self.send_device_command(clusters.WindowCovering.Commands.StopMotion())
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        await self.send_device_command(clusters.WindowCovering.Commands.UpOrOpen())
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        await self.send_device_command(clusters.WindowCovering.Commands.DownOrClose())
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Set the cover to a specific position."""
+        position = kwargs[ATTR_POSITION]
+        await self.send_device_command(
+            clusters.WindowCovering.Commands.GoToLiftPercentage(position)
+        )
+
+    async def send_device_command(self, command: Any) -> None:
+        """Send device command."""
+        await self.matter_client.send_device_command(
+            node_id=self._endpoint.node.node_id,
+            endpoint_id=self._endpoint.endpoint_id,
+            command=command,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        self._attr_current_cover_position = self.get_matter_attribute_value(
+            clusters.WindowCovering.Attributes.CurrentPositionLiftPercentage
+        )
+        LOGGER.info("GOT CURRENT POSITION %s", self._attr_current_cover_position)
+
+
+# Discovery schema(s) to map Matter Attributes to HA entities
+DISCOVERY_SCHEMAS = [
+    MatterDiscoverySchema(
+        platform=Platform.COVER,
+        entity_description=CoverEntityDescription(key="MatterCover"),
+        entity_class=MatterCover,
+        required_attributes=(
+            clusters.WindowCovering.Attributes.CurrentPositionLiftPercent100ths,
+            clusters.WindowCovering.Attributes.OperationalStatus,
+        ),
+        optional_attributes=(),
+        # restrict device type to prevent discovery in switch platform
```

Comment:
This is not what happens. It stops on off plug in unit device attributes from getting discovered as covers.

Reply:
I don't particularly understand the way this is handled, the matter cluster can get a definite device type, can't we use that information?

Reply:
You can but you have overlapping device types. This was just something you can ignore here, I believe you copied it from another platform ?

Reply:
I may need to familiarize myself with the implementation again.